com.unity.render-pipelines.core/Tests/Runtime/CollectionExtensionTests.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using System.Collections;
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace UnityEngine.Rendering.Tests
+{
+    public class CollectionExtensionTests
+    {
+        static TestCaseData[] s_ListTestsCaseDatas =
+        {
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 1, 2).SetName("Remove middle"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 0, 2).SetName("Remove front"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 0, 6).SetName("Remove all"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 5, 1).SetName("Remove back"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 5, -1).SetName("Count negative"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, -1, 2).SetName("Index negative"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 5, 0).SetName("Count 0"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 0, 0).SetName("Index 0"),
+            new TestCaseData(new List<int> {1,2,3,4,5,6}, 5, 5).SetName("Count exceeds list size"),
+        };
+
+        [Test, TestCaseSource(nameof(s_ListTestsCaseDatas))]
+        public void CheckRemoveRange(List<int> list, int startIndex, int count)
```

Comment:
Can you make a more explicit property test: a test that takes input and returns a bool (true on success)

Also it would be more manageable to separate test on the list and on simple list (the test currently fails if any of those property fails, so investigation is longer to make)


To clarify, I am thinking of

```csharp
// Even if it is long, it is a clear property description
public bool ItemInRangeAreRemovedAfterRemoveRange<TList, TValue>(TList list, int startm int count)
    where TList: IList<TValue>
{
    // Model based testing: model is List<TValue>.RemoveRange
    // Same test code as you have: copy, execute RemoveRange on both list, compare results and errors    
}

[Test, TestCaseSource(nameof(...))]
public void ItemInRangeAreRemovedAfterRemoveRangeForList(int[] ints , int startIndex, int count) { }

// Use same data input
[Test, TestCaseSource(nameof(...))]
public void ItemInRangeAreRemovedAfterRemoveRangeForSimpleList(int[] ints , int startIndex, int count) { }
```